File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.swift
func solution(_ n:Int) -> Int {
    var numA = 0
    var numB = 1
    var sum = 0
    
    if n == 0 { return 0 }

    for _ in 1..<n {
        sum = numA + numB % 1234567
        numA = numB % 1234567
        numB = sum % 1234567
    }
    
    return numB
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.js
function solution(n) {    
    let f0 = 0
    let f1 = 1
    
    for(let i = 2; i <= n; i++) {
        const next = (f0 + f1) % 1234567
        f0 = f1
        f1 = next
    }

    return n === 0 ? 0 : f1
}
